assemble/bin/accumulo-cluster

```diff
@@ -43,52 +43,74 @@ function invalid_args {
+  if [[ ! -f ${conf}/cluster.yml ]]; then
+    echo "ERROR: A 'cluster.yml' file was not found at ${conf}/cluster.yml"
+    echo "Please make sure it exists and is configured with the host information. Run 'create-config' to create an example configuration."
     exit 1
   fi
 
-  unset manager1
-  if [[ -f "${conf}/$manager_file" ]]; then
-    manager1=$(grep -Ev '(^#|^\s*$)' "${conf}/$manager_file" | head -1)
+  config=$(${accumulo_cmd} org.apache.accumulo.core.conf.cluster.ClusterConfigParser ${conf}/cluster.yml)
+
+  declare -A HOSTS
+
+  HOSTS["MANAGERS"]=$(grep managers $config | awk -F':' '{ print $2 }')
+  HOSTS["MONITORS"]=$(grep monitors $config | awk -F':' '{ print $2 }')
+  HOSTS["GC"]=$(grep gc $config | awk -F':' '{ print $2 }')
+  HOSTS["TRACER"]=$(grep tracer $config | awk -F':' '{ print $2 }')
+  HOSTS["TSERVERS"]=$(grep tservers $config | awk -F':' '{ print $2 }')
+
+  grep "compaction" $config $2>1 > /dev/null
+  compaction_in_config=$?
+
+  if [[ $compaction_in_config -eq 0 ]]; then
+    HOSTS["COMPACTION_COORDINATORS"]=$(grep "compaction.coordinators" $file | awk -F':' '{ print $2 }')
+    HOSTS["COMPACTION_QUEUES"]=$(grep "compaction.compactors.queues" $file | awk -F':' '{ print $2 }')
+
+    IFS_ORIG=$IFS
+    IFS=","
+    for queue in ${HOSTS["COMPACTION_QUEUES"]}; do
+      HOSTS["COMPACTORS_$queue"]=$(grep "compaction.compactors.$queue" $file | awk -F':' '{ print $2 }')
+    done
+    IFS=$IFS_ORIG
   fi
 
-  if [[ -z "${monitor}" ]] ; then
-    monitor=$manager1
-    if [[ -f "${conf}/monitor" ]]; then
-      monitor=$(grep -Ev '(^#|^\s*$)' "${conf}/monitor" | head -1)
-    fi
-    if [[ -z "${monitor}" ]] ; then
-      echo "Could not infer a Monitor role. You need to either define \"${conf}/monitor\","
-      echo "or make sure \"${conf}/$manager_file\" is non-empty."
-      exit 1
-    fi
+  if [[ ${#HOSTS["MANAGERS"]} -eq 0 ]]; then
+    echo "ERROR: managers not found in ${conf}/cluster.yml"
+    exit 1
   fi
-  if [[ ! -f "${conf}/tracers" ]]; then
-    if [[ -z "${manager1}" ]] ; then
-      echo "Could not find a manager node to use as a default for the tracer role."
-      echo "Either set up \"${conf}/tracers\" or make sure \"${conf}/$manager_file\" is non-empty."
-      exit 1
-    else
-      echo "$manager1" > "${conf}/tracers"
-    fi
+
+  if [[ ${#HOSTS["TSERVERS"]} -eq 0 ]]; then
+    echo "ERROR: tservers not found in ${conf}/cluster.yml"
+    exit 1
   fi
-  if [[ ! -f "${conf}/gc" ]]; then
-    if [[ -z "${manager1}" ]] ; then
-      echo "Could not infer a GC role. You need to either set up \"${conf}/gc\" or make sure \"${conf}/$manager_file\" is non-empty."
-      exit 1
-    else
-      echo "$manager1" > "${conf}/gc"
-    fi
+
+  unset manager1
+  manager1=$(echo "${HOSTS["MANAGERS"]}" | cut -d"," -f1)
+
+  if [[ ${#HOSTS["MONITORS"]} -eq 0 ]]; then
+    echo "WARN: monitors not found in ${conf}/cluster.yml, using first manager host $manager1"
+    HOSTS["MONITORS"]=$manager1
+    exit 1
+  fi
```

Comment:
You probably don't want to exit with an error if it's just a warning and you're mitigating it by assigning the first manager. Also, the Java code could probably do that logic, so that its output is always valid, and there's less checking for whether a server type is set or falls back to the default.